src/executor/plan_executor.cpp

```diff
@@ -147,6 +147,13 @@ void PlanExecutor::ExecutePlan(
   PELOTON_ASSERT(plan != nullptr && txn != nullptr);
   LOG_TRACE("PlanExecutor Start (Txn ID=%" PRId64 ")", txn->GetTransactionId());
 
+  if (static_cast<StatsType>(settings::SettingsManager::GetInt(
+      settings::SettingId::stats_mode)) != StatsType::INVALID) {
+    stats::BackendStatsContext::GetInstance()
+        ->GetQueryLatencyMetric()
+        .StartTimer();
```

Comment:
The Execute function in query.h has a comment saying that it is asynchronous. I think your code looks right, but please double check if this is correct.